Tests/Extractor/JsExtractorTest.php

```diff
@@ -0,0 +1,149 @@
+use Symfony\Component\Translation\MessageCatalogue;
+use Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory;
+use Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem;
+use Bazinga\Bundle\JsTranslationBundle\Extractor\JsExtractor;
+
+final class JsExtractorTest extends PHPUnit_Framework_TestCase
+{
+    const TEST_LOCALE = 'en';
+    const TEST_KEY_1 = 'test-key-1';
+    const TRANSLATION_PATH_PUBLIC = '/translation-path/public';
+    const TRANSLATION_PATH_VIEWS = '/translation-path/views';
+
+    /**
+     * @var JsExtractor
+     */
+    private $sut;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var FinderFactory
+     */
+    private $finderFactory;
+
+    /**
+     * @var MessageCatalogue
+     */
+    private $messageCatalogue;
+
+    private $folder;
+    private $fileName;
+    private $fileContent;
+
+    public function setUp()
+    {
+        $this->filesystem = $this->prophesize(Filesystem::class);
+        $this->finderFactory = $this->prophesize(FinderFactory::class);
+        $this->messageCatalogue = new MessageCatalogue(self::TEST_LOCALE);
+
+        $this->sut = new JsExtractor(
+            $this->filesystem->reveal(),
+            $this->finderFactory->reveal()
+        );
+    }
+
+    public function testExtractShouldRetrieveTransKey()
+    {
+        $this->givenASourceFolderWithATransFunctionUsage();
+        $this->thenTheFinderWillFindAJsFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransKeyIsInMessageCatalogue();
+    }
+
+    public function testExtractShouldRetrieveTransChoiceKey()
+    {
+        $this->givenASourceFolderWithATransChoiceFunctionUsage();
+        $this->thenTheFinderWillFindAJsFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransChoiceKeyIsInMessageCatalogue();
+    }
+
+    private function givenASourceFolderWithATransFunctionUsage()
+    {
+        $this->givenASourceFolder();
+
+        $this->fileContent = <<<STRING
+        Translator.trans('test-key-1');
+        Translator.trans();
+        Translator.trans(variable);
+STRING;
+    }
+
+    private function givenASourceFolderWithATransChoiceFunctionUsage()
+    {
+        $this->givenASourceFolder();
+
+        $this->fileContent = <<<STRING
+        Translator.transChoice('test-key-1', 5);
+        Translator.transChoice();
+        Translator.transChoice(variable, 5);
+STRING;
+    }
+
+    private function givenASourceFolder()
+    {
+        $this->folder = self::TRANSLATION_PATH_VIEWS;
+        $this->fileName = 'test.js';
+        $this->filesystem
+            ->exists(self::TRANSLATION_PATH_PUBLIC)
+            ->willReturn(true);
+    }
+
+    private function thenTheFinderWillFindAJsFile()
+    {
+        $finder = $this->prophesize(Finder::class);
+
+        $finder
+            ->files()
```

Comment:
please don't mock the finder. This does not allow you to ensure that it is configured properly, as you never ensure that the way you configured it does way you expect it to be. You should not mock what you don't own (if Symfony changes the behavior of the Finder, this test will not warn you about it, as nothing ensures that your mock describes the real behavior)